### ISSUE TITLE
"panic: runtime error: index out of range" when running glide up in container

### ISSUE THREAD

**Issue by krancour** [Masterminds/glide#149]
I've built glide into [this](https://hub.docker.com/r/krancour/deis-go-dev/) Docker image (source is [here](https://github.com/krancour/docker-go-dev)).  This image is _just_ the golang:1.5.1 image + glide.

When executing as follows:

```
docker run -it --rm -v ${PWD}:/app -w /app -e GO15VENDOREXPERIMENT=1 krancour/deis-go-dev:0.1.0 glide up
```

With the following trivial `glide.yaml`:

```
package: github.com/krancour/deis-v2-router-poc
import:
- package: k8s.io/kubernetes
  version: v1.1.1
  subpackage: pkg
```

The following is the result.  But note that when running this _outside_ the container, it works just fine.  Might this be a glide bug?  Or possibly a user error?

```
root@66826437c948:/app# glide up
[WARN] Unable to locate home directory
[INFO] Fetching updates for k8s.io/kubernetes.
[INFO] Setting version for k8s.io/kubernetes to v1.1.1.
[INFO] Found Godeps.json file.
[INFO] Found Godeps.json file for "k8s.io/kubernetes"
[INFO] Setting version for github.com/beorn7/perks to b965b613227fddccbfffe13eae360ed3fa822f8d.
[INFO] Setting version for github.com/cpuguy83/go-md2man to 71acacd42f85e5e82f70a55327789582a5200a90.
[INFO] Setting version for github.com/onsi/ginkgo to d981d36e9884231afa909627b9c275e4ba678f90.
[INFO] Setting version for github.com/prometheus/procfs to 490cc6eb5fa45bf8a8b7b73c8bc82a8160e8531d.
[INFO] Setting version for golang.org/x/crypto to c84e1f8e3a7e322d497cd16c0e8a13c7e127baf3.
[INFO] Setting version for golang.org/x/net to c2528b2dd8352441850638a8bb678c2ad056fd3e.
[INFO] Setting version for golang.org/x/tools to 4f50f44d7a3206e9e28b984e023efce2a4a75369.
[INFO] Setting version for github.com/appc/spec to c928a0c907c96034dfc0a69098b2179db5ae7e37.
[INFO] Setting version for k8s.io/heapster to 0e1b652781812dee2c51c75180fc590223e0b9c6.
[INFO] Setting version for google.golang.org/api to 0c2979aeaa5b573e60d3ddffe5ce8dca8df309bd.
[INFO] Setting version for github.com/fsouza/go-dockerclient to 76fd6c68cf24c48ee6a2b25def997182a29f940e.
[INFO] Setting version for github.com/kardianos/osext to 8fef92e41e22a70e700a96b29f066cda30ea24ef.
[INFO] Setting version for github.com/coreos/go-semver to 6fe83ccda8fb9b7549c9ab4ba47f47858bc950aa.
[INFO] Setting version for github.com/golang/groupcache to 604ed5785183e59ae2789449d89e73f3a2a77987.
[INFO] Setting version for speter.net/go/exp/math/dec/inf to 42ca6cd68aa922bc3f32f1e056e61b65945d9ad7.
[INFO] Setting version for github.com/ghodss/yaml to 73d445a93680fa1a78ae23a5839bad48f32ba1ee.
[INFO] Setting version for github.com/prometheus/client_model to fa8ad6fec33561be4280a8f0514318c79d7f6cb6.
[INFO] Setting version for github.com/samuel/go-zookeeper to 177002e16a0061912f02377e2dd8951a8b3551bc.
[INFO] Setting version for github.com/ugorji/go to 821cda7e48749cacf7cad2c6ed01e96457ca7e9d.
[INFO] Setting version for github.com/xyproto/simpleredis to 5292687f5379e01054407da44d7c4590a61fd3de.
[INFO] Setting version for golang.org/x/exp to d00e13ec443927751b2bd49e97dea7bf3b6a6487.
[INFO] Setting version for github.com/ClusterHQ/flocker-go to 3f33ece70f6571f0ec45bfae2f243ab11fab6c52.
[INFO] Setting version for github.com/coreos/go-oidc to ee7cb1fb480df22f7d8c4c90199e438e454ca3b6.
[INFO] Setting version for github.com/docker/spdystream to b2c3287865f3ad6aa22821ddb7b4692b896ac207.
[INFO] Setting version for github.com/inconshreveable/mousetrap to 76626ae9c91c4f2a10f34cad8ce83ea42c93bb75.
[INFO] Setting version for github.com/kr/pty to 05017fcccf23c823bfdea560dcc958a136e54fb7.
[INFO] Setting version for github.com/mitchellh/mapstructure to 740c764bc6149d3f1806231418adb9f52c11bcbf.
[INFO] Setting version for github.com/abbot/go-http-auth to c0ef4539dfab4d21c8ef20ba2924f9fc6f186d35.
[INFO] Setting version for github.com/spf13/pflag to 8e7dc108ab3a1ab6ce6d922bbaff5657b88e8e49.
[INFO] Setting version for google.golang.org/cloud to 2e43671e4ad874a7bca65746ff3edb38e6e93762.
[INFO] Setting version for gopkg.in/yaml.v2 to d466437aa4adc35830964cffc5b5f262c63ddcb4.
[INFO] Setting version for github.com/pborman/uuid to ca53cad383cad2479bbba7f7a1a05797ec1386e4.
[INFO] Setting version for github.com/golang/glog to 44145f04b68cf362d9c4df2182967c2275eaefed.
[INFO] Setting version for github.com/prometheus/client_golang to 3b78d7a77f51ccbc364d4bc170920153022cfd08.
[INFO] Setting version for github.com/vaughan0/go-ini to a98ad7ee00ec53921f08832bc06ecf7fd600e6a1.
[INFO] Setting version for golang.org/x/oauth2 to b5adcc2dcdf009d0391547edc6ecbaff889f5bb9.
[INFO] Setting version for github.com/golang/protobuf to 7f07925444bb51fa4cf9dfe6f7661876f8852275.
[INFO] Setting version for github.com/gorilla/context to 215affda49addc4c8ef7e2534915df2c8c35c6cd.
[INFO] Setting version for github.com/coreos/go-systemd to 97e243d21a8e232e9d8af38ba2366dfcfceebeba.
[INFO] Setting version for github.com/daviddengcn/go-colortext to b5c0891944c2f150ccc9d02aecf51b76c14c2948.
[INFO] Setting version for github.com/elazarl/go-bindata-assetfs to 3dcc96556217539f50599357fb481ac0dc7439b9.
[INFO] Setting version for github.com/emicklei/go-restful to 1f9a0ee00ff93717a275e15b30cf7df356255877.
[INFO] Setting version for github.com/google/gofuzz to bbcb9da2d746f8bdbd6a936686a0a6067ada0ec5.
[INFO] Setting version for github.com/russross/blackfriday to 77efab57b2f74dd3f9051c79752b2e8995c8b789.
[INFO] Setting version for github.com/scalingdata/gcfg to 37aabad69cfd3d20b8390d902a8b10e245c615ff.
[INFO] Setting version for github.com/syndtr/gocapability to 2c00daeb6c3b45114c80ac44119e7b8801fdd852.
[INFO] Setting version for github.com/Sirupsen/logrus to 51fe59aca108dc5680109e7b2051cbdcfa5a253c.
[INFO] Setting version for github.com/coreos/go-etcd to 4cceaf7283b76f27c4a732b20730dcdb61053bf5.
[INFO] Setting version for github.com/gorilla/mux to 8096f47503459bcc74d1f4c487b7e6e42e5746b5.
[INFO] Setting version for github.com/imdario/mergo to 6633656539c1639d9d78127b7d47c622b5d7b6dc.
[INFO] Setting version for github.com/influxdb/influxdb to afde71eb1740fd763ab9450e1f700ba0e53c36d0.
[INFO] Setting version for github.com/jonboulle/clockwork to 3f831b65b61282ba6bece21b91beea2edc4c887a.
[INFO] Setting version for github.com/rackspace/gophercloud to f92863476c034f851073599c09d90cd61ee95b3d.
[INFO] Setting version for github.com/appc/cni to 2a58bd9379ca33579f0cf631945b717aa4fa373d.
[INFO] Setting version for github.com/dgrijalva/jwt-go to 5ca80149b9d3f8b863af0e2bb6742e608603bd99.
[INFO] Setting version for github.com/skynetservices/skydns to 1be70b5b8aa07acccd972146d84011b670af88b4.
[INFO] Setting version for github.com/spf13/cobra to 68f5a81a722d56241bd70faf6860ceb05eb27d64.
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Masterminds/glide/vendor/github.com/Masterminds/vcs.NewHgRepo(0xc8203f2a80, 0x22, 0xc8203f26c0, 0x26, 0xc8203f2a01, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/vcs/hg.go:42 +0x549
github.com/Masterminds/glide/vendor/github.com/Masterminds/vcs.NewRepo(0xc8203f2a80, 0x22, 0xc8203f26c0, 0x26, 0x0, 0x0, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/vcs/repo.go:155 +0x25a
github.com/Masterminds/glide/yaml.(*Dependency).GetRepo(0xc8202bc6e0, 0xc8203f26c0, 0x26, 0x0, 0x0, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/yaml/yaml.go:177 +0x12e
github.com/Masterminds/glide/cmd.VcsVersion(0xc8202bc6e0, 0xc8204ba8e0, 0xb, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/cmd/get_imports.go:534 +0x38e
github.com/Masterminds/glide/cmd.flattenGlideUp(0xc8200c8d78, 0xc820119280, 0x1d, 0x0, 0x0, 0xc800000000, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/cmd/flatten.go:159 +0x787
github.com/Masterminds/glide/cmd.recFlatten(0xc8200c8d78, 0xc820466b00, 0x0, 0x0, 0xc820000000, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/cmd/flatten.go:115 +0x705
github.com/Masterminds/glide/cmd.Flatten(0x7fbaa08795b8, 0xc8200f5140, 0xc8200240a0, 0x0, 0x0, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/cmd/flatten.go:58 +0x12ef
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).doCommand(0xc820118ac0, 0xc8200f5cb0, 0x7fbaa08795b8, 0xc8200f5140, 0x0, 0x0, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:209 +0x86
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).runRoute(0xc820118ac0, 0x938ef8, 0x6, 0x7fbaa08795b8, 0xc8200f5140, 0x938e00, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:164 +0x4c9
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).HandleRequest(0xc820118ac0, 0x938ef8, 0x6, 0x7fbaa08795b8, 0xc8200f5140, 0xc8200c9000, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:131 +0x2fc
main.setupHandler(0xc82014e000, 0x938ef8, 0x6, 0x7fbaa08795b8, 0xc8200f5140, 0xc820118ac0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:469 +0x48c
main.commands.func10(0xc82014e000)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:400 +0x89f
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.Command.Run(0x938ef8, 0x6, 0x938ef0, 0x2, 0xc820111ef0, 0x1, 0x1, 0x9b8e80, 0x2a, 0xa2bbe0, ...)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.(*App).Run(0xc820069d40, 0xc82000a0c0, 0x2, 0x2, 0x0, 0x0)
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:175 +0xfd6
main.main()
    /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:119 +0x784

goroutine 368 [IO wait]:
net.runtime_pollWait(0x7fbaa087aae0, 0x72, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820431aa0, 0x72, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820431aa0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820431a40, 0xc8200f9000, 0x1000, 0x1000, 0x0, 0x7fbaa0875050, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820024538, 0xc8200f9000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc82043df20, 0x7fbaa087aea0, 0xc820024538, 0x5, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82020c000, 0xa1ff17, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc82020c000, 0xc820017000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fbaa0887878, 0xc82020c000, 0xc8203fb4f8, 0xc820017000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82044a5e0, 0xc820017000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82030a0c0)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82030a0c0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8203fb4a0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:685 +0xc78

goroutine 33 [IO wait, 2 minutes]:
net.runtime_pollWait(0x7fbaa087aa20, 0x72, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820430920, 0x72, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820430920, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8204308c0, 0xc820128400, 0x400, 0x400, 0x0, 0x7fbaa0875050, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820024120, 0xc820128400, 0x400, 0x400, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8203f3410, 0x7fbaa087aea0, 0xc820024120, 0x5, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200aa000, 0xa1ff17, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200aa000, 0xc8200de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fbaa0887878, 0xc8200aa000, 0xc8200d9e98, 0xc8200de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820452b40, 0xc8200de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82030be00)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82030be00, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200d9e40)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:685 +0xc78

goroutine 34 [select, 2 minutes]:
net/http.(*persistConn).writeLoop(0xc8200d9e40)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 40 [IO wait, 2 minutes]:
net.runtime_pollWait(0x7fbaa087a960, 0x72, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82034a290, 0x72, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82034a290, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82034a230, 0xc820129400, 0x400, 0x400, 0x0, 0x7fbaa0875050, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820024218, 0xc820129400, 0x400, 0x400, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc820366120, 0x7fbaa087aea0, 0xc820024218, 0x5, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200aa2c0, 0xa1ff17, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200aa2c0, 0xc82040f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fbaa0887878, 0xc8200aa2c0, 0xc8203facb8, 0xc82040f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8204335a0, 0xc82040f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820012180)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820012180, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8203fac60)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:685 +0xc78

goroutine 41 [select, 2 minutes]:
net/http.(*persistConn).writeLoop(0xc8203fac60)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 47 [IO wait, 2 minutes]:
net.runtime_pollWait(0x7fbaa087aba0, 0x72, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82034baa0, 0x72, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82034baa0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82034ba40, 0xc8200a1000, 0x400, 0x400, 0x0, 0x7fbaa0875050, 0xc82000e140)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200242b8, 0xc8200a1000, 0x400, 0x400, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8202ba480, 0x7fbaa087aea0, 0xc8200242b8, 0x5, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200aa580, 0xa1ff17, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200aa580, 0xc820345000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fbaa0887878, 0xc8200aa580, 0xc8203fb448, 0xc820345000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820416e80, 0xc820345000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820013c80)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820013c80, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8203fb3f0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:685 +0xc78

goroutine 369 [select]:
net/http.(*persistConn).writeLoop(0xc8203fb4a0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 48 [select, 2 minutes]:
net/http.(*persistConn).writeLoop(0xc8203fb3f0)
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
    /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:686 +0xc9d
```

**Comment by krancour** [comment]
fwiw, this was with version 0.7.2

I've been able to address the issue with the home dir not being found by adding `-e GLIDE_HOME=/root` to the `docker run`

That didn't change my result, so I think we can rule out that as a factor.

Subsequently, I also have verified that this issue also exists in master.

**Comment by krancour** [comment]
Oddly, switching distros seems to fix this.

Changing my Dockerfile to begin: `FROM golang:1.5.1-alpine` (and then a few other changes to account for what isn't already in there) got me moving again.

**Comment by krancour** [comment]
Scratch that.  The issue still occurs even after changing distros.  When I swapped distros, my new image was initially missing mercurial.  I was succeeding only because with mercurial absent, the offending bits of code didn't execute.  After I added mercurial to the image, I'm right back to having the problem.

I am now noticing a pattern that every time this fails, it does fail fetching a package that uses mercurial as the VCS.

**Comment by krancour** [comment]
Specifically, it seems to be failing when transitively vendoring `bitbucket.org/ww/goautoneg`.

**Comment by krancour** [comment]
And here is why... within a container, assuming my local worspace has been mounted into the container, when `hg paths` is run on `bitbucket.org/ww/goautoneg`, this is what happens:

```
/go/src/github.com/krancour/deis-v2-router-poc/vendor/bitbucket.org/ww/goautoneg # hg paths
not trusting file /go/src/github.com/krancour/deis-v2-router-poc/vendor/bitbucket.org/ww/goautoneg/.hg/hgrc from untrusted user 1000, group 50
not trusting file /go/src/github.com/krancour/deis-v2-router-poc/vendor/bitbucket.org/ww/goautoneg/.hg/hgrc from untrusted user 1000, group 50
```

And that doesn't match the regex in Masterminds/vcs.

**Comment by krancour** [comment]
So, ultimately, that's a permissions issue... not at all a glide bug.  Closing.

**Comment by mattfarina** [comment]
@krancour thanks for sharing this issue. Glide shouldn't panic but instead share an error. So, I'm going re-open so we can clean up the experience.

**Comment by dnephin** [comment]
I just hit this as well, using git (not hg), running in a container, as root.  I was able to work around it by running `chown root:root -R vendor` in the container. I don't fully understand why it's a problem when files are owned by a non-root user, since root would be able to read/write them.

**Comment by mattfarina** [comment]
@dnephin what version of Glide are you using and how are you getting it? What's the image this is running in? And, do you have an example project I could test?

Glide drops down to the VCS tool, in this case git, to handle its operations. Is the user running Glide root or a different user?

**Comment by dnephin** [comment]
I started on 0.6.2, then upgraded to 0.8.2. I got a similar panic with both versions. The base image is `golang:1.5` with just glide installed. The project directory (including `vendor/`) is mounted as a volume to the host `-v $PWD:/go/src/github.com/dnephin/project`

I unfortunately don't have an example project to test at this time. I believe what I had done was `chown daniel:daniel -R vendor` at some earlier time, then when I came back to it and ran `glide up` I  was getting the panic midway through the run while it was working on transitive dependencies.

I'll see if I can reproduce it with a minimal example.

**Comment by dnephin** [comment]
https://gist.github.com/dnephin/8046cca0b87e2d1b62ee

I've included the stack trace and the necessary files to reproduce it. Note that the first run is successful, it's on the second run, after doing a `chown`, that it panics.  I guess it was actually in the vcs HG calls.

**Comment by mattfarina** [comment]
@dnephin thanks!

**Comment by mikeln** [comment]
I'm getting a similar issue.  I have created a golang1.6 (+glide 0.10.2) container.   Doing a glide install with kubernetes as a dependency:
```
[DEBUG] Package github.com/prometheus/common/expfmt imports bitbucket.org/ww/goautoneg
[DEBUG] Missing bitbucket.org/ww/goautoneg. Trying to resolve.
[INFO] Fetching bitbucket.org/ww/goautoneg into /go/src/github.com/samsung-cnct/barrel/vendor
panic: runtime error: index out of range
```

I do not appear to have file permission issues.  Initial assessment is that this may be the first dependency that uses hg  vs git.   This dependency is autodetected by glide (i.e. I have not specifically stated it in glide.yam).  I also suspect this may be an SSH vs HTTPS issue, but I don't have any debug to show what command is being run.   

By hand in the container, I do know that hg clone <ssh path> does not work, but hg clone <https path> does work.  What does glide use by default?

**Comment by mikeln** [comment]
ok, my bad, I do see the same issue:
```
oot@de56a551c717:/go/src/github.com/samsung-cnct/barrel/vendor/bitbucket.org/ww/goautoneg# hg paths
not trusting file /go/src/github.com/samsung-cnct/barrel/vendor/bitbucket.org/ww/goautoneg/.hg/hgrc from untrusted user 1000, group staff
not trusting file /go/src/github.com/samsung-cnct/barrel/vendor/bitbucket.org/ww/goautoneg/.hg/hgrc from untrusted user 1000, group staff
```

**Comment by mikeln** [comment]
So, the error I saw, had to do with SSH (in hg) requires the key file to be owned by the user running glide.  On os-x the user running in the Docker container is not the same as the filesystem owner (i.e. the user doing the docker run).  I added a script before running a golang+glide container that queried the filesystem UID/GID, and used that to set the uid:gid of the apparent user running inside of the container.
